### ISSUE TITLE
Beartype 0.10.2 bug: beartype.vale doesn't accept @beartype -decorated functions

### ISSUE THREAD

**Issue by dycw** [beartype/beartype#104]
Hi @leycec , me reporting another bug again.

```python
from contextlib import suppress
from typing import Annotated
from typing import Any

from beartype import beartype
from beartype.vale import Is


_predicate_given_as_lambda = lambda x: x >= 0


@beartype
def this_works(x: Annotated[int, Is[_predicate_given_as_lambda]]) -> int:
    return x


def test_1() -> None:
    assert this_works(0) == 0


########


def _predicate_given_as_undecorated_func(x: Any, /) -> bool:
    return x >= 0


@beartype
def this_works_too(
    x: Annotated[int, Is[_predicate_given_as_undecorated_func]]
) -> int:
    return x


def test_2() -> None:
    assert this_works_too(0) == 0


########


@beartype
def _predicate_given_as_decorated_func(x: Any, /) -> bool:
    return x >= 0


# @beartype
# def this_doesnt_work(
#     x: Annotated[int, Is[_predicate_given_as_decorated_func]]
# ) -> int:
#     return x


def test_3() -> None:
    with suppress(NameError):
        assert this_doesnt_work(0) == 0
```

Uncomment the 5 lines to reveal:

```
____________________ ERROR collecting src/tests/test_foo.py ____________________
src/tests/test_foo.py:19: in <module>
    def this_doesnt(x: Annotated[int, Is[_non_negative]]) -> int:
.venv/lib/python3.9/site-packages/beartype/vale/_is/_valeis.py:315: in __getitem__
    return BeartypeValidator(
.venv/lib/python3.9/site-packages/beartype/vale/_core/_valecore.py:213: in __init__
    die_unless_func_args_len_flexible_equal(
.venv/lib/python3.9/site-packages/beartype/_util/func/arg/utilfuncargtest.py:131: in die_unless_func_args_len_flexible_equal
    raise exception_cls(
E   beartype.roar.BeartypeValeSubscriptionException: Callable <function _predicate_given_as_decorated_func at 0x7f8ff68c7430> flexible argument count 0 != 1.
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.12s ===============================
"""
```

**Comment by leycec** [comment]
So this is what it feels like when you fail to properly test yourself against yourself.

![it do be like that](https://c.tenor.com/Lu2BctabWxAAAAAM/face-palm-shake-my-head.gif)

Because this issue is embarrassing, we have no recourse but to patch this up in a new `0.10.3` patch release. Your fearless work ethic suggests to me that you never sleep. I am grateful for your sacrifice and look forward to you immediately breaking `beartype 0.10.3` across one knee, too.

If I sat on a gilded throne of crypto, I would give it all to you in honour of your tireless dedication to `0.10.x` release churn. Alas, all I have is this nostalgic muppets GIF.

**Comment by dycw** [comment]
Hi @leycec , I can only thank you for making the world's Python code bases a little better. Our codebase thanks you 2880 times to be precise.

```
> rg "@beartype" --stats
2882 matches
2882 matched lines
257 files contained matches
6291 files searched
60736 bytes printed
108346278 bytes searched
0.063399 seconds spent searching
0.145649 seconds
```

**Comment by leycec** [comment]
**Resolved by 4e91224.** Someday, @dycw. I promise you that someday the madness of the `0.10.x` release cycle shall end.

**Comment by leycec** [comment]
**Released with `beartype 0.10.3`.** As always, `pip` is your buddy:

```bash
$ pip3 install --upgrade beartype
```

Yet again, thanks so much for the gauntlet of *outstanding* issues. You've done a tremendous service for @beartype bros everywhere, @dcyw. This honey pot's for you. :honey_pot:
